include/s3select.h

```diff
@@ -831,16 +830,17 @@ void push_logical_predicate::operator()(s3select* self, const char* a, const cha
 void push_negation::operator()(s3select* self, const char* a, const char* b) const
 {
   std::string token(a, b);
-  base_statement* pred;
-  if (self->getAction()->condQ.empty() == false)
+  base_statement* pred{nullptr};
+
+  if (!self->getAction()->condQ.empty())
   {
     pred = self->getAction()->condQ.back();
     self->getAction()->condQ.pop_back();
   }
   //upon NOT operator, the logical and arithmetical operators are "tagged" to negate result.
   if (dynamic_cast<logical_operand*>(pred))
   {
-    logical_operand* f = S3SELECT_NEW(self, logical_operand, pred);
+    logical_operand* f = S3SELECT_NEW(self, logical_operand, pred); // todo: marked as "empty statemant"
```

Comment:
as discussed, it's not empty statement (it allocate)
probably wrong message.

Reply:
will remove the comment